xline/src/storage/kv_store.rs

```diff
@@ -937,6 +944,33 @@ mod test {
         Ok(())
     }
 
+    #[tokio::test(flavor = "multi_thread", worker_threads = 10)]
+    async fn test_kv_store_atomic() {
+        let db = DB::open(&StorageConfig::Memory).unwrap();
+        let store = Arc::new(init_store(Arc::clone(&db)).await.unwrap());
+        let _handle = tokio::spawn({
+            let store = Arc::clone(&store);
+            async move {
+                for i in 0..100_u8 {
+                    let req = RequestWithToken::new(
+                        PutRequest {
+                            key: "foo".into(),
+                            value: vec![i],
+                            ..Default::default()
+                        }
+                        .into(),
+                    );
+                    let (_res, ops) = store.after_sync(&req).await.unwrap();
+                    db.flush_ops(ops).unwrap();
+                }
+            }
+        });
+        tokio::time::sleep(std::time::Duration::from_micros(500)).await;
+        let revs = store.index.get_from_rev(b"foo", b"", 1);
+        let kvs = store.get_values(&revs).unwrap();
+        assert_eq!(kvs.len(), revs.len());
```

Comment:
Please add some concrete diagnostic messages for this assertion. Additionally, I don't get the point why this test can prove that the kv store process is atomic.

Reply:
Fixed, If the kv store is not atomic, some kvs maybe insert to index, but not flush to db, so `revs.len()` will lager than `kvs.len()`